### DISCUSSION TITLE
ack vs response

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#4709] [Q&A]
If I have two channels, channel A sending to channel B, how can I return a response from channel B so that it shows up in the response viewer in channel A.  I'm assuming the ack returned is something different than the response. 

I currently have a channel that sends to a state vaccination registry.  I get an ack back but I also get a response message back. I'm trying to replicate that with two channels in mirth to test some transforms.

**Comment by tonygermano** [comment]
In channel B, on the Source tab, under the Response drop down, you can select a destination from channel B. That destination's response will be returned to the caller (channel A.)

The description of the Response drop down from the official User Guide is:

> Determines what response to send back to the originating system. You may choose a destination's response, the post-processor return value, a response map variable, an auto-generation option, or None indicating that you don't want to send a response back at all.
> 
> Select **Auto-generate** to send a response generated by the inbound data type using the raw message:
> 
> - **Before processing**: Response generated before the channel processes the message (SENT status)
> - **After source transformer**: Response generated after the channel processes the message (source connector status)
> - **Destinations completed**: Response generated after the channel processes the message, with a status based on the destination statuses, using a precedence of _ERROR_ , _QUEUED_ , _SENT_ , _FILTERED_ .

If you are using the open source version of mirth with no custom plugins, the only inbound data type that supports auto-generating a response is HL7v2. If you have an auto-generate option selected with any other inbound data type, it will be no different than if you had selected **None**.
